### PR TITLE
Oidc split 2 apis

### PR DESCRIPTION
**Pull request by nessiLahav**
#### What does this PR do?
Create a stopping point for splitting the OIDC authenticators for 2 APIs
#### What ticket does this PR close?
Closes [relevant GitHub issues, eg #76]
#### Has the Version and Changelog been updated?

> Can we make a blog post, video, or animated GIF of this?

> Has this change been documented (Readme, docs, etc.)?

> Does the knowledge base need an update?

### CODE REVIEW

File: app/domain/authentication/strategy.rb [hunk 1]
```diff
@@ -206,6 +228,15 @@ def new_token(input)
       )
     end
 
+    def new_oidc_token(user_details)
```
Comment (orenbm) [inline app/domain/authentication/strategy.rb]: add todo comment explaining that this should be done in the new OidcTokenFactory and this factory should be used as a dependency in the new OidcStrategy

File: app/domain/authentication/strategy.rb [hunk 2]
```diff
@@ -206,6 +228,15 @@ def new_token(input)
...
+      # TODO: encrypt the id_token
+      AuthnOidc::OidcToken.new(
+        id_token_encrypted: user_details.id_token,
+        user_info: user_details.user_info.preferred_username,
```
Comment (orenbm) [inline app/domain/authentication/strategy.rb]: change to username

File: app/domain/authentication/strategy.rb [hunk 3]
```diff
@@ -110,6 +110,22 @@ def conjur_token(input)
     # Or take a different approach that accomplishes the same goals
     #
     def conjur_token_oidc(input)
+      oidc_token = oidc_token(input)
```
Comment (orenbm) [inline app/domain/authentication/strategy.rb]: add todo with validation & decryption

Comment (orenbm) [reply app/domain/authentication/strategy.rb]: in GetOidcToken

File: app/domain/authentication/authn_oidc/get_user_details.rb [hunk 4]
```diff
@@ -45,7 +45,8 @@ def user_details
           id_token: id_token,
```
Comment (orenbm) [inline app/domain/authentication/authn_oidc/get_user_details.rb]: Rename UserDetails

File: app/domain/authentication/authn_oidc/get_oidc_token.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    # dependencies: get key for decryption
+    # input: request_body type: raw text,  contains jwt_signed {id_token_encrypted + expiration_time + user_info }
+    # actions:
+    # - decode body
+    # - Validate input
+    # - Validate JWT signing token
+    # - Decrypt ID token
+    GetOidcToken = CommandClass.new(
+      dependencies: { },
+      # dependencies: { fetch_key: ::Conjur::FetchRequiredKey.new },
```
Comment (orenbm) [inline app/domain/authentication/authn_oidc/get_oidc_token.rb]: remove

File: app/domain/authentication/authn_oidc/get_user_details.rb (hunk 4 above)
Comment (nessiLahav) [reply app/domain/authentication/authn_oidc/get_user_details.rb]: OidcToken changed to OidcConjurToken

UserDetails changed to OidcIDTokenDetails

File: app/domain/authentication/strategy.rb (hunk 3 above)
Comment (nessiLahav) [reply app/domain/authentication/strategy.rb]: Exists inside the function

File: app/domain/authentication/strategy.rb (hunk 1 above)
Comment (nessiLahav) [reply app/domain/authentication/strategy.rb]: Done

File: app/domain/authentication/strategy.rb (hunk 2 above)
Comment (nessiLahav) [reply app/domain/authentication/strategy.rb]: done

File: app/domain/authentication/strategy.rb (hunk 3 above)
Comment (nessiLahav) [reply app/domain/authentication/strategy.rb]: done

File: app/domain/authentication/authn_oidc/get_user_details.rb (hunk 4 above)
Comment (nessiLahav) [reply app/domain/authentication/authn_oidc/get_user_details.rb]: done

File: app/domain/authentication/authn_oidc/get_oidc_token.rb (hunk 5 above)
Comment (nessiLahav) [reply app/domain/authentication/authn_oidc/get_oidc_token.rb]: done

**Pull request merged**